utils/kubernetes/expose/expose.go

```diff
@@ -0,0 +1,521 @@
+package expose
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/layer5io/meshkit/logger"
+	appsv1 "k8s.io/api/apps/v1"
+	appsv1beta1 "k8s.io/api/apps/v1beta1"
+	appsv1beta2 "k8s.io/api/apps/v1beta2"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	extensionsv1beta1 "k8s.io/api/extensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+)
+
+// Config is a the struct for Expose configuration
+type Config struct {
+	// Type is the type of the service
+	// could be either ClusterIP, NodePort
+	// or LoadBalancer. If an empty is string
+	// is provided then "ClusterIP" type service
+	// is created
+	Type string
```

Comment:
Are we considering for `ExternalName`?

Reply:
@leecalcote, I did not consider `ExternalName` for expose services and neither I have tested them, should I add that to besides `LoadBalancer`, `ClusterIP` and `NodePort`?

Reply:
Ok... let's skip as a potential future use case.